lib/models/custom_layers/custom_layer.dart

```diff
@@ -0,0 +1,21 @@
+import 'dart:async';
+import 'dart:developer';
+import 'dart:math' as math;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_map/flutter_map.dart';
+import 'package:trufi_core/blocs/custom_layer/custom_layers_cubit.dart';
+
+import 'package:latlong/latlong.dart';
+
+abstract class CustomLayer {
+  final String id;
+  CustomLayersCubit customLayersCubit;
+  Function onRefresh;
+  CustomLayer(this.id);
```

Comment:
Would it make sense to add here the status of the layer (boolean) and so that the state is not getting polluted with a Map<String, bool>?